libraries/node-core-library/src/Executable.ts

```diff
@@ -0,0 +1,419 @@
+    //
+    // See these articles for a full analysis:
+    // http://www.windowsinspired.com/understanding-the-command-line-string-and-arguments-received-by-a-windows-program/
+    // http://www.windowsinspired.com/how-a-windows-programs-splits-its-command-line-into-individual-arguments/
+
+    const environment: NodeJS.ProcessEnv = options && options.environment
+      || process.env;
+    const fileExtension: string = path.extname(resolvedPath);
+
+    if (fileExtension.toUpperCase() === '.JS') {
+      // Inherit the parent's node.exe
+      const nodePath: string = process.execPath;
+      const nodeArgs: string[] = [];
+      nodeArgs.push(resolvedPath);
+      nodeArgs.push(...args);
+      return child_process.spawnSync(nodePath, nodeArgs, spawnOptions);
+    }
+
+    if (os.platform() === 'win32') {
+      // Do we need a custom handler for this file type?
+      switch (fileExtension.toUpperCase()) {
+        case '.EXE':
+        case '.COM':
+          // okay to execute directly
+          break;
+        case '.BAT':
+        case '.CMD':
+          {
+            Executable._validateArgsForWindowsShell(args);
+
+            // These file types must be invoked via the Windows shell
+            let shellPath: string | undefined = environment.COMSPEC;
+            if (!shellPath || !Executable._canExecute(shellPath, context)) {
+              shellPath = Executable.tryResolve('cmd.exe');
+            }
+            if (!shellPath) {
+              throw new Error(`Unable to execute "${path.basename(resolvedPath)}" `
+                + `because CMD.exe was not found in the PATH`);
+            }
+
+            const shellArgs: string[] = [];
+            shellArgs.push('/d');
+            shellArgs.push('/s');
+            shellArgs.push('/c');
+            shellArgs.push(resolvedPath);
+            shellArgs.push(...args);
+
+            return child_process.spawnSync(shellPath, shellArgs, spawnOptions);
+          }
+        default:
+          throw new Error(`Cannot execute "${path.basename(resolvedPath)}" because the file type is not supported`);
+      }
+    }
+
+    return child_process.spawnSync(resolvedPath, args, spawnOptions);
+  }
+
+  /**
+   * Search for an executable file with the specified name and return its path.
+   *
+   * @remarks
+   * If the name has a relative path, it will be resolved relative to the current working directory.
+   * If the name has no path, then the shell PATH will be searched.  If the name is missing a Windows
+   * file extension, it may be appended (based on the PATHEXT environment variable).
+   *
+   * @param name - the name of the executable, which may be missing the path or file extension
+   * @param options - optional other parameters
+   * @returns the absolute path of the executable, or undefined if it was not found
+   */
+  public static tryResolve(name: string, options?: IExecutableResolveOptions): string | undefined {
```

Comment:
>tryResolve [](start = 16, length = 10)

Document that the `name` parameter is just the executable name/path and cannot contain args.